python/lsst/analysis/ap/nb_utils.py

```diff
@@ -273,3 +275,82 @@ def compare_sources(butler1, butler2, query1, query2,
         _ = unique2.pop('pathexists')
 
     return unique1, unique2, matched
+
+
+def get_xy_from_source_table(table, wcs, degrees=False):
+    """Convert ra/dec coordinates in an astropy table/pandas data frame to
+    pixel x/y positions.
+    """
+    ra = table['coord_ra']
+    dec = table['coord_dec']
+    x, y = wcs.skyToPixelArray(ra, dec, degrees=degrees)
+    return astropy.table.Table.from_pandas(pd.DataFrame({'x': x, 'y': y}))
+
+
+def display_images(butler, visit, detector, backend="firefly"):
+    """Display the science/template/difference images for a given
+    visit+detector, with sources and mask planes overlaid.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        Butler to load data from.
+    visit, detector : `int`
+        Visit and detector ids to load data for.
+    backend : str, optional
+        afw display backend to display to (typically "firefly" or "ds9").
```

Comment:
Are there any other things necessary to use `ds9`? When I tried using it with `nb_utils.display_images(butler, 2025050100309, 22, backend="ds9")`, I got nine `XPA$ERROR no 'xpaset' access points match template: ds9` messages followed by an `AttributeError: Device ds9 has no attribute "alignImages"`.

Reply:
Ah, yes, I didn't add `alignImages` for ds9 because I didn't know how. I've asked on DM-50875 about Gregory's thoughts on that.

Reply:
Followup here: I couldn't get it to work in ds9, and didn't want to spend more time on it. I've made it at least not fail out in this function, and print a warning if that functionality is not available.